python/ray/serve/multiplex.py

```diff
@@ -0,0 +1,90 @@
+from ray._private.async_compat import sync_to_async
+from collections import OrderedDict
+from typing import Any, Callable
+import logging
+from ray.serve._private.constants import SERVE_LOGGER_NAME
+import inspect
+import asyncio
+
+
+logger = logging.getLogger(SERVE_LOGGER_NAME)
+
+
+class _ModelMultiplexWrapper:
+    def __init__(
+        self,
+        model_load_func: Callable[[str], Any],
+        self_arg: Any,
+        max_num_models_per_replica: int,
+    ):
+        """Initialize the model multiplexer.
+
+        The model multiplexer is a wrapper class that wraps the model load function
+        and provides the LRU caching functionality, and the model load function should
+        be a coroutine function that takes the model ID as the first argument and
+        returns the model handle.
+        The model multiplexer will also ensure that the number of models on the current
+        replica does not exceed the specified limit.
+        The model will be unloaded in the LRU order, the model multiplexer will call the
+        model's __del__ attribute if it exists to clean up the model resources eagerly.
+
+        Args:
+            model_load_func: the model load async function.
+            self_arg: self argument when model_load_func is class method.
+            max_num_models_per_replica: the maximum number of models to be loaded on the
+                current replica. If it is -1, there is no limit for the number of models
+                per replica.
+        """
+        self.models = OrderedDict()
+        self._func = model_load_func
+        self.self_arg = self_arg
+        self.max_num_models_per_replica = max_num_models_per_replica
+
+    async def load_model(self, model_id: str) -> Any:
+        """Load the model if it is not loaded yet, and return the model handle.
+
+        Args:
+            model_id: the model ID.
+
+        Returns:
+            The model handle.
```

Comment:
"handle" is not meaningful here and in fact is extremely confusing because it makes it sound like it returns a `ServeHandle`
```suggestion
        Returns:
            The user-constructed model object.
```